### PR TITLE
boards/nrf52840dongle: Add spi feature and default pins

### PR DESCRIPTION
**Pull request by LordTy**
### Contribution description
The board does support SPI, however the feature was not enables.
I needed it for a project. 
The default pins are in a bit of a weird location (0.14 is on the bottom of the board).
But it is accessible.

### Testing procedure
Compile a project that uses SPI for the nrf52840dongle.

### CODE REVIEW

File: boards/nrf52840dongle/Makefile.features [hunk 1]
```diff
@@ -4,6 +4,7 @@ CPU_MODEL = nrf52840xxaa
 FEATURES_PROVIDED += periph_pwm
 FEATURES_PROVIDED += periph_uart
 FEATURES_PROVIDED += periph_usbdev
+FEATURES_PROVIDED += periph_spi
```
Comment (benpicco) [inline boards/nrf52840dongle/Makefile.features:7]: The board's Kconfig will need an update too

**Comment by benpicco** [comment]
AFAIR the SPI function can be mapped to any pin, so if you find another configuration more convenient, we can also use that instead.

File: boards/nrf52840dongle/Makefile.features (hunk 1 above)
Comment (LordTy) [reply boards/nrf52840dongle/Makefile.features:7]: Ow, I did not know that that existed.
I do not know how to test the Kconfig, but I think it is added now.

**Comment by LordTy** [comment]
I am already using the default pins in my current project.
As the convenient location is specific to the use case, I think users should change this when needed.

File: boards/nrf52840dongle/Makefile.features (hunk 1 above)
Comment (benpicco) [reply boards/nrf52840dongle/Makefile.features:7]: It's tested by `tests/kconfig_features` - but this is a trivial change to begin with.

Please squash!

**Review by benpicco — APPROVED** [review]
looks good, please squash

**Pull request merged**